Add chess pieces with STI

Mikeysax opened the pull request:
Had issues with STI because I was running into a rubocop documentation error and I was trying to inherit from ChessPiece model which didn't exist yet. Should be okay now.

lasley commented:
This is the STI ticket? Please try to make the PR title a short description of the feature being added so I can relate to a ticket

Additionally, we've got some merge conflicts that will need to be resolved before merge. Let me know when you've done that and I'll review :smile:

Mikeysax commented:
@lasley I'm trying to see what conflicts there are but I'm not sure where to look. Gonna fix them in a moment. Sorry about the naming, I will remember that for future.

lasley commented:
Best way is to just start a merge from master to this branch. Git will walk you through the rest - https://help.github.com/articles/resolving-a-merge-conflict-from-the-command-line/

lasley commented:
(In a public repo you would actually rebase this branch on top of master in order to avoid the merge entry in our commit history. I don't know if I'd worry about that here, just something to be aware of. https://www.atlassian.com/git/tutorials/merging-vs-rebasing/ )

Mikeysax commented:
@lasley Fixed schema conflict

Mikeysax commented:
@lasley ah interesting, I will take a look at that!

lasley commented on app/models/chess_piece.rb:
```diff
@@ -1,4 +1,3 @@
-# Model for each chess piece in a game
 class ChessPiece < ActiveRecord::Base
```
Adding the chess prefix may be a bit redundant with the application context already being chess. Not a big deal, just a thought.

lasley commented on app/views/layouts/application.html.erb:
```diff
@@ -5,17 +5,92 @@
+          <img src="https://raw.githubusercontent.com/thoughtbot/refills/master/source/images/twitter-logo-circle.png" alt="Twitter">
+        </a></li>
+        <li><a href="javascript:void(0)">
+          <img src="https://raw.githubusercontent.com/thoughtbot/refills/master/source/images/youtube-logo-circle.png" alt="YouTube">
```
May want to look into Font Awesome icons instead of these static images:
* http://fortawesome.github.io/Font-Awesome/icon/youtube/
* https://github.com/bokmann/font-awesome-rails

Mikeysax replied:
Yeah, I'm gonna change them.

lasley commented on app/views/static_pages/index.html.erb:
```diff
@@ -1 +1 @@
-<h1>Firechess</h1>
+<h1>This is the index</h1>
```
Merge issue? Seems Firechess is a better title.

lasley commented on db/migrate/20160116040219_create_games.rb:
```diff
@@ -1,7 +1,7 @@
   def change
     create_table :games do |t|
-      t.timestamps null: false
+      t.timestamps
```
`null: false` was the better option here IMO

lasley commented:
What is the proposed strategy of for determining the specific piece type that is being used?

Mikeysax commented:
@lasley I'm not sure what you mean by the last comment, can you explain?

Mikeysax commented:
@lasley I used those awesome icons you linked, refactored the footer. It's not perfect but useable for now. I also changed the schema back to null: false for timestamps. I'm alittle confused about the last comment because I thought setting the STI up initially would only involve linking the chesspiece model to the subclass models? We would add the logic later?

Mikeysax commented:
@lasley I'm having a issue with the javascript and the header. When ever I click the logo it opens the menu in mobile view?

lasley commented:
@Mikeysax 

Yup actually don't worry about the last comment, I had a brain fart.

To clarify on the issue- when you are in mobile view and click the header, the menu toggles?

lasley commented on app/views/layouts/application.html.erb:
```diff
@@ -5,17 +5,93 @@
+      <div class="navigation-wrapper">
+        <a href="javascript:void(0)" class="navigation-menu-button" id="js-mobile-menu">MENU</a>
+        <nav role="navigation">
+          <%= link_to 'Firechess', root_path, :class => 'logo' %>
```
This is your logo clicking issue, needs to be outside the nav role

Mikeysax replied:
I moved it outside on my localmachine and it still is bugging. Maybe it's the script?

Mikeysax replied:
@lasley When I'm in the mobile view, if I've clicked the logo or anything linked to root_path the mobile menu opens automatically and is unresponsive when I try to close it. It's just frozen open and I'm not really sure.

Mikeysax replied:
Also, I can't seem to push the sign in and sign up to the right side even with implementing the grid system.

Mikeysax commented:
@lasley There is still that bug with the header opening after clicking a link to root path but other than that it is okay I think.

Mikeysax commented:
@lasley I rebased a few of the last commits. I think everything looks good?

lasley commented:
@Mikeysax - If you click the `root_path

Mikeysax commented:
@lasley Yeah, no matter what is linked, if it's linked to the root_path it will expand the header when you minimize to mobile view.

lasley commented:
@Mikeysax - 

Well that's weird. I'm not familiar with Bourbon, but that header seems fine in regards to construction. You sure that the header doesn't just start expanded, and that it's actually the link?

Would you like me to merge with this issue still present?

Mikeysax commented:
@lasley Yeah I think there isn't much we can do right now to fix it. When I click a link to the root_path it will open up the menu in mobile view and you can't close it unless you refresh. Otherwise it's fine.

lasley commented:
Alright, go ahead and make a ticket to fix this issue so it doesn't get forgotten
